packages/types/src/json.ts

```diff
@@ -443,3 +454,42 @@ export interface DeletedObjectJSON {
 
 export type SignInFirstFactorJSON = CamelToSnake<SignInFirstFactor>;
 export type SignInSecondFactorJSON = CamelToSnake<SignInSecondFactor>;
+
+/**
+ * Types for WebAuthN passkeys
+ */
+
+type Base64UrlString = string;
+
+interface PublicKeyCredentialUserEntityJSON {
+  name: string;
+  displayName: string;
+  id: Base64UrlString;
+}
+
+export interface ExcludedCredentialJSON {
+  type: 'public-key';
+  id: Base64UrlString;
+  transports?: ('ble' | 'hybrid' | 'internal' | 'nfc' | 'usb')[];
+}
+
+interface AuthenticatorSelectionCriteriaJSON {
+  requireResidentKey: boolean;
+  residentKey: 'discouraged' | 'preferred' | 'required';
+  userVerification: 'discouraged' | 'preferred' | 'required';
+}
+
+export interface PublicKeyCredentialCreationOptionsJSON {
+  rp: PublicKeyCredentialRpEntity;
```

Comment:
This type comes from typescript, do you think we should ship our own ?

Reply:
I'd prefer keeping it as-is so when someone searches for it on Google they'll get relevant results. No reason to alias something if we don't add value

Reply:
I'm worrying about types changing in a feature version of typescript. Not sure how often they update their "web api" types.